### PR TITLE
Feature/moz 342 create list campaign

### PR DESCRIPTION
**Pull request by dnaluz**
You will need to create a MailChimp account for this PR.  You'll need to create an API key as well and save it in the theme settings.  There are also a couple new theme settings that need to be saved for this to work

Screen Shot 2020-02-24 at 11 11 40 AM

A small caveat with the MailChimp account you will need to delete the original audience that gets created when you sign up to your account.  This is because the free MailChimp account only allows for 1 Audience(list).  So before you create new campaigns to test, you will need to delete the list because of the 1 list limit associated with a free MailChimp account.  Let me know if you have any questions regarding the MailChimp integration.

### CODE REVIEW

**Comment by dnaluz** [comment]
@kycapstick 

OK so this now has all of the campaign functionality merged into one branch.  We should test this before merging into develop!

File: buddypress/members/single/profile.php [hunk 1]
```diff
@@ -268,6 +268,54 @@
...
+            $campaigns = get_user_meta($user->ID, 'campaigns', true);
+            $campaign_count = 0;
+        ?>
+        <?php if(is_array($campaigns) && sizeof($campaigns) > 0): ?>
+        <h2 class="profile__heading"><?php print __("Campaigns Participated In", "community-portal"); ?></h2>
+        <div class="profile__card">
+        <?php foreach($campaigns AS $campaign_id): ?>
+        <?php 
+            $campaign = get_post($campaign_id);
+            $description = get_field('card_description', $campaign->ID);
```
Comment (kycapstick) [inline buddypress/members/single/profile.php]: This line and the next three are throwing the following error: "Trying to get property 'ID' of non-object"

**Comment by kycapstick** [comment]
@dnaluz everything looks great, only thing I noticed is that there's some extra whitespace at the bottom if a user is only subscribed to one campaign
Screen Shot 2020-03-03 at 11 28 40 AM

File: buddypress/members/single/profile.php (hunk 1 above)
Comment (dnaluz) [reply buddypress/members/single/profile.php]: Hmmm, what I'm assuming here is you deleted a campaign but the IDs are still being stored.  I added a check to make sure there is an object still.

**Comment by kycapstick** [comment]
@dnaluz this looks good for the errors I was getting, but I'm still seeing the extra white space at the bottom of the user's campaigns

**Comment by dnaluz** [comment]
> @dnaluz this looks good for the errors I was getting, but I'm still seeing the extra white space at the bottom of the user's campaigns

I don't think this is a case of extra space I think what is happening here is you deleted a campaign and it's still being referenced within the user data but since it's been deleted it can show any information.

I'll need to refactor

**Comment by dnaluz** [comment]
@kycapstick refactored some subscribing and unsubscribing code.  If you could retest this that would be great.  Make sure everything shows up on the mailchimp side as well!

**Comment by kycapstick** [comment]
Functionality looks good! Only thing I noticed is if I subscribe then unsubscribe, the text on the subscribe button changes to "Participate" even though I'd set it to be "subscribe" in the ACF fields.

**Comment by dnaluz** [comment]
@kycapstick Just pushed a fix for the copy issue

**Pull request merged**